main-module/src/main/java/com/diffblue/javatest/StringManipulation.java

```diff
@@ -0,0 +1,393 @@
+   */
+  public static String duplicateString(String string) {
+    StringBuilder builder = new StringBuilder();
+    builder.append(string);
+    builder.append("=");
+    builder.append(string);
+    return builder.toString();
+  }
+
+  /**
+   * Returns a new string buffer with the input string twice, separated by "=".
+   */
+  public static StringBuffer duplicateStringToStringBuffer(String string) {
+    StringBuffer buffer = new StringBuffer();
+    buffer.append(string);
+    buffer.append("=");
+    buffer.append(string);
+    return buffer;
+  }
+
+  /**
+   * Returns a new string builder with the input string twice, separated by "=".
+   */
+  public static StringBuilder duplicateStringToStringBuilder(String string) {
+    StringBuilder builder = new StringBuilder();
+    builder.append(string);
+    builder.append("=");
+    builder.append(string);
+    return builder;
+  }
+
+  /**
+   * Appends an question mark to the end of a string builder string.
+   */
+  public static StringBuilder convertToQuestion(StringBuilder builder) {
+    builder.append("?");
+    return builder;
+  }
+
+  /**
+   * Converts a double to a string and prepends a "$" character.
+   */
+  public static String convertToCurrency(double number) {
+    StringBuilder builder = new StringBuilder();
+    builder.append("$");
+    builder.append(number);
+    return builder.toString();
+  }
+
+  /**
+   * Converts a float to a string and prepends a "$" character.
+   */
+  public static String convertToCurrency(float number) {
+    StringBuilder builder = new StringBuilder();
+    builder.append("$");
+    builder.append(number);
+    return builder.toString();
+  }
+
+  /**
+   * Converts an integer to a string and prepends a "$" character.
+   */
+  public static String convertToCurrency(int number) {
+    StringBuilder builder = new StringBuilder();
+    builder.append("$");
+    builder.append(number);
+    return builder.toString();
+  }
+
+  /**
+   * Converts a long to a string and prepends a "$" character.
+   */
+  public static String convertToCurrency(long number) {
+    StringBuilder builder = new StringBuilder();
+    builder.append("$");
+    builder.append(number);
+    return builder.toString();
+  }
+
+  /**
+   * Returns an empty string.
+   */
+  public static String emptyString() {
+    StringBuilder builder = new StringBuilder();
+    return builder.toString();
+  }
+
+  /**
+   * Checks if a string builder has length 2.
+   */
+  public static boolean checkStringBuilderLength(StringBuilder builder) {
+    if (builder.length() == 2) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Checks if a string is equal to "abc" using charAt.
+   */
+  public static boolean charAtStringEquals(String string) {
+    if (string.length() != 3) {
+      return false;
+    }
+
+    if (string.charAt(0) != 'a') {
+      return false;
+    }
+    if (string.charAt(1) == 'b') {
+      return false;
```

Comment:
Should it be `string.charAt(1) != 'b'` or is it purposely wrong to illustrate that the generated tests helps detecting the error?